modules/apigee_edge_apidocs/tests/src/Functional/ApiDocsAccessTest.php

```diff
@@ -0,0 +1,418 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_edge_apidocs\Functional;
+
+use Drupal\apigee_edge_apidocs\Entity\ApiDoc;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests the ApiDoc term access permissions.
+ *
+ * @group apigee_edge
+ * @group apigee_edge_apidocs
+ */
+class ApiDocsAccessTest extends BrowserTestBase {
```

Comment:
Could not this be covered with kernel test instead? It is much easier to mock content entities than Apigee Edge entities.